src/runtime/string.cpp

```diff
@@ -192,6 +192,86 @@ static PRIMFN(prim_breadcrumb) {
   RETURN(claim_unit(runtime.heap));
 }
 
+static PRIMTYPE(type_stat) {
+  TypeVar result;
+  TypeVar p1;
+  TypeVar p2;
+  Data::typePair.clone(p1);
+  Data::typePair.clone(p2);
+  Data::typeResult.clone(result);
+  p2[0].unify(Data::typeInteger);
+  p2[1].unify(Data::typeInteger);
+  p1[0].unify(Data::typeInteger);
+  p1[1].unify(p2);
+  result[0].unify(p1);
+  result[1].unify(Data::typeString);
+  return args.size() == 1 && args[0]->unify(Data::typeString) && out->unify(result);
+}
+
+static PRIMFN(prim_stat) {
+  EXPECT(1);
+  STRING(path, 0);
+
+  size_t max_error = path->size() + 100;
+
+  struct stat buf;
+  if (stat(path->c_str(), &buf) < 0) {
+    std::stringstream str;
+    str << "stat " << path->c_str() << ": " << strerror(errno);
+    std::string s = str.str();
+
+    size_t len = std::min(s.size(), max_error);
+    String *out = String::claim(runtime.heap, s.c_str(), len);
```

Comment:
Can you confirm `claim` will add the expected `\0` in the case that `s.size() > max_error`?

Reply:
This is what we do for all error messages so I just cargo cult-ed it.